src/html_scanner.zig

```diff
@@ -0,0 +1,297 @@
+const std = @import("std");
```

Comment:
whenever a file with a single `struct` is added, make that struct top-level and make the file name the name of that struct (`html_scanner.zig` to `HTMLScanner.zig`). when i do these, i then make one of the first line the text `const StructName = @This()` directly followed by the fields.

this pattern pairs really well with imports at the end of the file, so the start reads well

Reply:
There are about 5 places we do this currently

I'm not opposed to it

but these patterns are really only good when they're consistent